server/static/dashboard/collab.js

```diff
@@ -7,42 +10,486 @@ collab.fetch(function(err) {
+    $(regexElt).addClass(classesToAdd);
+    $(afterRegexElt).addClass(classesToAdd);
+
+    regexElt.classList.add('diff-regex');
+
+    beforeRegexElt.appendChild(document.createTextNode(
+      text.substring(endOfLastRegex, match.indexInLine)));
+    regexElt.appendChild(document.createTextNode(
+      text.substring(match.indexInLine, match.indexInLine + match.length)));
+    afterRegexElt.appendChild(document.createTextNode(
+      text.substring(match.indexInLine + match.length)));
+
+    if (endOfLastRegex > 0) {
+      // Need to remove the last child, since the three elts
+      // created here represent the same characters as the last child
+      newElts.pop();
+    }
+
+    newElts.push(beforeRegexElt);
+    newElts.push(regexElt);
+    newElts.push(afterRegexElt);
+
+    // Increment index of last regex so we know where to split
+    // if there's another regex in the element
+    endOfLastRegex = match.indexInLine + match.length;
+
+  });
+
+  return newElts;
+}
+
+/**
+ * Replace {node}'s current children with the children
+ *   in {newChildNodes}.
+ */
+function replaceChildren(node, newChildNodes) {
+  // Remove the old children
+  while (node.firstChild) {
+    node.removeChild(node.firstChild);
+  }
+  // Add the new children
+  newChildNodes.forEach(function(child) {
+    node.appendChild(child);
+  });
+}
+
+/**
+ * Get the Ajax URL that returns the total diff for {filepath}
+ *   using {threshold}.
+ */
+function getAjaxUrlForTotalDiff(filepath, threshold) {
+  var url = '/ops/' + project + '/' + collabid + '/' + filepath
+    + (cutoff ? '?cutoff=' + cutoff : '')
+    + (threshold ? (cutoff ? '&threshold=' + threshold
+                           : '?threshold=' + threshold)
+                 : '');
+  return url;
+}
+
+/**
+ * Creates and returns the DOM for a total diff visualization
+ *   where the deleted code is shown on the right.
+ *   (for visual1_deletesOnSide).
+ * Adds that DOM inside {node}.
+ */
+function addTotalDiffDeletesOnSideDom(diff, node) {
+  var divNormal = document.createElement('div');
+  divNormal.classList.add('div-normal');
+  divNormal.classList.add('col-xs-6');
+  var divDeleted = document.createElement('div');
+  divDeleted.classList.add('div-deleted');
+  divDeleted.classList.add('col-xs-6');
+
+  diff.forEach(function(part) {
+    if (part.value.length > 0) {
+    
+      var elt = document.createElement('span');
+      // Save part data for hide common prefix
+      $(elt).data('part', part);
+
+      if (part.added) {
+        elt.classList.add('span-added');
+      } else if (part.removed) {
+        elt.classList.add('span-removed');
+        if (part.original) {
+          elt.classList.add('span-original');
+        }
+      } else {
+        elt.classList.add('span-original');
+      }
+
+      elt.appendChild(document.createTextNode(part.value));
+      divNormal.appendChild(elt);
+
+      elt2 = elt.cloneNode(true);
+      $(elt2).data('part', part);
```

Comment:
Shouldn't the cloned DOM already have this? Try using `dataset` instead of jQuery's `data`

Reply:
Done.